walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -270,5 +272,154 @@ public void notEnoughRoom() {
 
     Assert.assertEquals(expected, actual);
   }
+
+  @Test
+  public void optionalAttendeeAllDay() {
+    // Each person has different events and an optional attendee has an all day event.
+    // We should see three options because the all day event is an optional attendee's event, 
+    // so we only return time slots that work for mandatory attendees.
+    //
+    // Events  :       |--A--|     |--B--|
+    //           |--------------C--------------| optional
+    // Day     : |-----------------------------|
+    // Options : |--1--|     |--2--|     |--3--|
+
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 1", TimeRange.fromStartDuration(TIME_0800AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 2", TimeRange.fromStartDuration(TIME_0900AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_B)),
+        new Event("Event 3", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, 
+            TimeRange.END_OF_DAY, true), Arrays.asList(PERSON_C)));
+
+    MeetingRequest request =
+        new MeetingRequest(Arrays.asList(PERSON_A, PERSON_B), DURATION_30_MINUTES);
+    request.addOptionalAttendee(PERSON_C);
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
+            TimeRange.fromStartEnd(TIME_0830AM, TIME_0900AM, false),
+            TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true));
+
+    Assert.assertEquals(expected, actual);
```

Comment:
Structure tests into 'arrange', 'act', 'assert' blocks

https://docs.google.com/document/d/1WqjdTF5ifkBx-9Ls3T1V4reAdnh0DRe_WZcvYlmnBlk/edit?hl=en#heading=h.yiwzzgckpwke


As you can see tests can get very long and complicated so just having this kind of breakdown is useful for whoever reads your code next.

Reply:
Since you have a single line break between `Collection<Event> events` and `MeetingRequest request` then the `arrange` and `act` block must be separated by multiple lines to make the division clear.

Alternatively (but less recommended) you can also add `// Arrange`, '// Act`, and `// Assert` comments to divide the unit test.
```
Collection<Event> events = Arrays.asList(
        new Event("Event 1", TimeRange.fromStartDuration(TIME_0800AM, DURATION_30_MINUTES),
            Arrays.asList(PERSON_A)),
        new Event("Event 2", TimeRange.fromStartDuration(TIME_0900AM, DURATION_30_MINUTES),
            Arrays.asList(PERSON_B)),
        new Event("Event 3", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, 
            TimeRange.END_OF_DAY, true), Arrays.asList(PERSON_C)));

MeetingRequest request =
        new MeetingRequest(Arrays.asList(PERSON_A, PERSON_B), DURATION_30_MINUTES);
    request.addOptionalAttendee(PERSON_C);

<----------two line break here-------->
Collection<TimeRange> actual = query.query(events, request);

<----------two line break here-------->
Collection<TimeRange> expected =
        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
            TimeRange.fromStartEnd(TIME_0830AM, TIME_0900AM, false),
            TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true));

Assert.assertEquals(expected, actual);
```




Anothe rthing is this three block thing might not necessarily be needed if the unit test is very small, in which case youmight be able to collapse act and assert into one block.